src/modules/roc_core/target_darwin/roc_core/format_tid.cpp

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <pthread.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "roc_core/format_tid.h"
+
+namespace roc {
+namespace core {
+
+bool format_tid(char* buf, size_t bufsz) {
+    uint64_t tid = 0;
+    pthread_threadid_np(NULL, &tid);
+    if (snprintf(buf, bufsz, "%llu", tid) < 0)
+        return false;
+    return true;
```

Comment:
I'm not sure if uint64_t is guaranteed to be unsigned long long. I suggest to add an explicit cast:

```
snprintf(buf, bufsz, "%llu", (unsigned long long)tid)
```

Also, the snprintf error checking is not quite correct. It should be something like:

```
int ret = snprintf(buf, bufsz, "%llu", tid);
return ret > 0 && (size_t)ret < bufsz;
```

Reply:
The same for linux version.